File: src/test/java/de/roamingthings/workbench/testcontainers/WelcomePage.java
package de.roamingthings.workbench.testcontainers;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;

public class WelcomePage extends PageObject {

    @FindBy(id = "name")
    private WebElement nameField;

    @FindBy(id = "submit")
    private WebElement submitButton;

    public WelcomePage(WebDriver driver) {
        super(driver);
    }

    public boolean isInitialized() {
        return submitButton.isDisplayed();
    }

    public GreetingPage submit(){
        submitButton.click();
        return new GreetingPage(driver);
    }

    public void enterName(String name) {
        nameField.clear();
        nameField.sendKeys(name);
    }

    public String name() {
        return nameField.getAttribute("value");
    }

}

File: src/test/kotlin/de/roamingthings/workbench/testcontainers/kt/WelcomePage.kt
package de.roamingthings.workbench.testcontainers.kt

import org.openqa.selenium.WebDriver
import org.openqa.selenium.WebElement
import org.openqa.selenium.support.FindBy

class WelcomePage(driver: WebDriver) : PageObject(driver) {

    @FindBy(id = "name")
    private val nameField: WebElement? = null

    @FindBy(id = "submit")
    private val submitButton: WebElement? = null

    val isInitialized: Boolean
        get() = submitButton!!.isDisplayed

    fun submit(): GreetingPage {
        submitButton!!.click()
        return GreetingPage(driver)
    }

    fun enterName(name: String) {
        nameField!!.clear()
        nameField.sendKeys(name)
    }

    fun name(): String {
        return nameField!!.getAttribute("value")
    }

}
